src/domain/tradingview/DataFeed.ts

```diff
@@ -0,0 +1,431 @@
+  }
+
+  subscribeBars(
+    symbolInfo: LibrarySymbolInfo,
+    resolution: ResolutionString,
+    onTick: SubscribeBarsCallback,
+    listenerGuid: string
+  ): void {
+    const token = getTokenBySymbol(this.chainId, symbolInfo.name);
+    const isStable = token.isStable;
+
+    const visualMultiplier = parseInt(symbolInfo.unit_id ?? "1");
+
+    const interval = new PauseableInterval<Bar | undefined>(
+      async ({ lastReturnedValue }) => {
+        let candlesToFetch = 1;
+
+        const currentCandleTime = getCurrentCandleTime(SUPPORTED_RESOLUTIONS_V2[resolution]);
+
+        if (lastReturnedValue) {
+          const periodSeconds = RESOLUTION_TO_SECONDS[resolution];
+
+          const diff = Math.abs(currentCandleTime - lastReturnedValue.time);
+          if (diff >= periodSeconds) {
+            candlesToFetch = Math.ceil(diff / periodSeconds);
+          }
+        }
+
+        let prices: FromOldToNewArray<Bar> = [];
+
+        try {
+          prices = !isStable
+            ? await this.fetchCandles(symbolInfo.name, resolution, candlesToFetch)
+            : this.getStableCandles(currentCandleTime, resolution, candlesToFetch);
+        } catch (e) {
+          // eslint-disable-next-line no-console
+          console.error(e);
+          return lastReturnedValue;
+        }
+
+        let newLastReturnedValue: Bar | undefined = lastReturnedValue;
+
+        let didPatchPreviousCandle = false;
+
+        for (const price of prices) {
+          if (lastReturnedValue?.time && price.time < lastReturnedValue.time) {
+            continue;
+          }
+
+          if (lastReturnedValue?.time && price.time > lastReturnedValue.time && !didPatchPreviousCandle) {
+            didPatchPreviousCandle = true;
+            const previousBarWithNewClose = {
+              ...lastReturnedValue,
+              close: price.open,
+              low: Math.min(lastReturnedValue.low, price.open),
+              high: Math.max(lastReturnedValue.high, price.open),
+            };
+
+            onTick(multiplyBarValues(formatTimeInBarToMs(previousBarWithNewClose), visualMultiplier));
+          }
+
+          onTick(multiplyBarValues(formatTimeInBarToMs(price), visualMultiplier));
+
+          newLastReturnedValue = price;
+        }
+
+        return newLastReturnedValue;
+      },
+      this.tradePageVersion === 1 ? V1_UPDATE_INTERVAL : V2_UPDATE_INTERVAL
+    );
+
+    this.subscriptions[listenerGuid] = interval;
+  }
+
+  unsubscribeBars(listenerGuid: string): void {
+    this.subscriptions[listenerGuid].destroy();
+    delete this.subscriptions[listenerGuid];
+  }
+
+  onReady(callback: OnReadyCallback): void {
+    callback({
+      supported_resolutions: Object.keys(
+        this.tradePageVersion === 1 ? SUPPORTED_RESOLUTIONS_V1 : SUPPORTED_RESOLUTIONS_V2
+      ) as ResolutionString[],
+      supports_marks: false,
+      supports_timescale_marks: false,
+      supports_time: true,
+      // @ts-ignore
+      reset_cache_timeout: 100,
```

Comment:
Is it an issue with TV so it doesn't know about this field?

Reply:
at first i decided to keep this legacy undocumented field, but i removed it, lets see if it breaks anything